device/device.go

```diff
@@ -461,10 +461,10 @@ func (device *Device) BindSetMark(mark uint32) error {
 	// clear cached source addresses
 	device.peers.RLock()
 	for _, peer := range device.peers.keyMap {
-		peer.Lock()
-		defer peer.Unlock()
-		if peer.endpoint != nil {
-			peer.endpoint.ClearSrc()
+		ep, _ := peer.endpoint.Load().(conn.Endpoint)
+		if ep != nil {
+			ep.ClearSrc()
```

Comment:
We should document on the interface that ClearSrc must be internally synchronized

Reply:
Good catch. I started proposing a way to synchronize endpoint src clears on `Peer` that makes the load, clear, store race less confusing, but then realized it's still subject to the same issue down in the endpoint itself.

Maybe we could at least put this dance in one func with better docs.

Reply:
What if we never mutate endpoint internally in-place, and instead we copy-on-write when we want to clear its source? There’s already a copy created per rx packet when wireguard-go is in standalone mode, and source clearing is not that frequent. Our tailscaled endpoint implementation doesn’t need to actually copy (return self), and we already synchronize it internally, and ClearSrc() is no-op.